File: Lab 08/LabCheckOut8-1.py
# Python code template for lab 08, EE 021
import numpy as np

# 1a. Create an empty array that has three rows and three columns. 
# Print the array.
num_rows = 3
num_cols = 3
################################################################################
# The empty function initializes an empty array, and it contains random values.
my_array = np.empty([num_rows,num_cols]) # replace None with your actual code.
################################################################################
print(my_array)



#################################################################################
# 1b. Fill up the array with all zeros. Must use for loops for this task.
# Hint: Use two nested for loops to iterate over the rows and columns of the array.
# Hint: Use the range function to create an iterable sequence of 
# numbers from 0 to the number of rows or columns.

for i in range(num_rows):
    for j in range(num_cols):
        my_array[i][j] = 0
#################################################################################





#################################################################################
# 1c. Update the values at all indices of the array with 
# values equal to the following 3 x 3 matrix: 
# Print the updated array with a \t (tab) separator between elements.
# [-1, 0, 1],
# [-2, 0, 2],
# [-1, 0, 1]
# You must update the already existing my_array variable, not create a new one.
my_array[0][0] = -1
my_array[0][2] = 1
my_array[1][0] = -2
my_array[1][2] = 2
my_array[2][0] = -1
my_array[2][2] = 1
#################################################################################


#################################################################################
# Print the updated array with a \t (tab) separator between elements.
print("Updated array is")
for i in range(num_rows):
    for j in range(num_cols):
        print(my_array[i][j], end = '\t')
    print()

File: Lab 08/LabCheckOut8-1.cpp
// C++ code template for lab check out 08, EE 021

#include <iostream>

using namespace std;

int main() {
    ////////////////////////////////////////////////////////////////////////////
    // 2a. Create an empty array that has two rows and two columns.
    // Print the array.
    const int num_rows = 2;
    const int num_cols = 2;
    // Declare my_array in the next line with the given number of rows and columns.
    // The line below declares a 2x2 array. Fix this line to declare a num_rows x num_cols array.
    int my_array[num_rows][num_cols]; // Fix this with correct variable reference for rows and columns


    ////////////////////////////////////////////////////////////////////////////

    // Print the array, code provided to you. 
    // Make sure that you understand the use of for loops to print the array.
    cout << "Empty array:" << endl;
    for (int i = 0; i < num_rows; i++) {
        for (int j = 0; j < num_cols; j++) {
            cout << my_array[i][j] << "\t";
        }
        cout << endl;
    }

    ////////////////////////////////////////////////////////////////////////////
    // 2b. Fill up the array with all zeros. Must use for loops for this task.
    // Similar to 1a, you may use loops to fill up the array with zeros.
    

    for (int i = 0; i < num_rows; i++) {
        for (int j = 0; j < num_cols; j++) {
            // Fill up the array with zeros.
            my_array[i][j] = 0;
        }
    }
    //////////////////////  //////////////////////////////////////////////////////
    // 2c. Update the values at all indices of the array with 
    // values equal to the following 3 x 3 matrix: 
    // [7, 9],
    // [1, 5] 
    // You must update the already existing my_array variable, not create a new one.

    my_array[0][0] = 7;
    my_array[0][1] = 9;
    my_array[1][0] = 1;
    my_array[1][1] = 5;
    ////////////////////////////////////////////////////////////////////////////


    ////////////////////////////////////////////////////////////////////////////
    // Print the updated array with a \t (tab) separator between elements.
    cout << "Updated array:" << endl;
    for (int i = 0; i < num_rows; i++) {
        for (int j = 0; j < num_cols; j++) {
            cout << my_array[i][j] << "\t";
        }
        cout << endl;
    }
    ////////////////////////////////////////////////////////////////////////////
}